lib/puppet/gettext/config.rb

```diff
@@ -10,19 +10,45 @@ module Puppet::GettextConfig
   # Used instead of features because we initialize gettext before features is available.
   # Stubbing gettext if unavailable is handled in puppet.rb.
   begin
-    require 'gettext-setup'
+    require 'fast_gettext'
     require 'locale'
+
+    # Make translation methods (e.g. `_()` and `n_()`) available everywhere.
+    class ::Object
+      include FastGettext::Translation
+    end
+
     @gettext_loaded = true
   rescue LoadError
+    # TODO do we want to add the stubs here, or leave it up to the caller
+    # to decide how to handle the failure case?
     @gettext_loaded = false
   end
 
-  # Whether we were able to require gettext-setup and locale
-  # @return [Boolean] true if gettext-setup was successfully loaded
+  # Whether we were able to require fast_gettext and locale
+  # @return [Boolean] true if translation gems were successfully loaded
   def self.gettext_loaded?
     @gettext_loaded
   end
 
+  def self.translations_loaded?(project_name)
+    if @loaded_repositories[project_name]
+      return true
+    else
+      return false
+    end
+  end
+
+  def self.create_text_domain(domain_name)
```

Comment:
This will probably end up being redone, and the API will probably changed, once we decide how we want to proceed with the per environment stuff.